grpc/noncebalancer/noncebalancer.go

```diff
@@ -88,19 +107,22 @@ func (p *Picker) Pick(info balancer.PickInfo) (balancer.PickResult, error) {
 	// Get the destination prefix from the RPC context.
 	destPrefixVal := info.Ctx.Value(nonce.PrefixCtxKey{})
 	if destPrefixVal == nil {
-		// This should never happen.
+		// This should never happen unless the client failed to set the prefix
+		// in the RPC context.
 		return balancer.PickResult{}, errMissingPrefixCtxKey
 	}
 	destPrefix, ok := destPrefixVal.(string)
 	if !ok {
-		// This should never happen.
+		// This should never happen unless the client set the prefix in the RPC
+		// context with the wrong type.
 		return balancer.PickResult{}, errInvalidPrefixCtxKeyType
 	}
 
 	sc, ok := p.prefixToBackend[destPrefix]
 	if !ok {
-		// No backend SubConn was found for the destination prefix.
-		return balancer.PickResult{}, balancer.ErrNoSubConnAvailable
+		// No backend SubConn was found for the destination prefix. Return an
+		// error so the client can retry with a new nonce.
```

Comment:
```suggestion
		// No backend was found for the destination prefix. Most likely
		// this is because the backend that issued the nonce was restarted. This
		// happens right after backend restarts, or when ACME clients redeem
		// nonces that were issued hours ago.
```

Changed from explaining what should happen next to explaining "why" this happens. The error itself explains what should happen next.